codalab/worker/local_run/local_run_state.py

```diff
@@ -275,7 +275,7 @@ def _transition_from_PREPARING(self, run_state):
             message = "Cannot assign enough resources: %s" % str(e)
             logger.error(message)
             logger.error(traceback.format_exc())
-            return run_state._replace(stage=LocalRunStage.CLEANING_UP, failure_message=message)
+            return run_state._replace(run_status=message)
```

Comment:
Why this change?

Reply:
This change will ensure jobs won't fail when there are not enough computing resources that could be assigned. This was changed two months ago from non-failing to failing state. But please let me know if we want to fail it on the worker side.

Reply:
This case shouldn't happen as the bundle manager really shouldn't be assigning jobs to workers that can't run them.  If we just kick it back, then won't we just end up in a loop?

Reply:
Yes, I agree that we shouldn't assign jobs to a worker if there are not enough resources that can be allocated to run it. However, this piece of logic has been there for a long time and I guess it would be good to have a secondary check from the worker side to prevent any error from bundle manager so that we keep this logic here? 
Removing the `stage=LocalRunStage.CLEANING_UP` will let bundle stay in `preparing` stage. Once there are enough resources to run, the bundle will move to `running` stage gain.

Reply:
So job just stays in `preparing` on that worker?  Can the user kill the job in `preparing`?

Reply:
User can definitely `kill` the job in `preparing` if they want. I think the key is to decide if a job should stay alive if there aren't enough computing resources that could be found on the worker side.  `run_status` will be updated with a message similar to `Cannot assign enough resources: Requested more GPUs (10) than available (0 currently out of 0 on the machine)`.